statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,480 @@
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+        replicates : integer
+            The number of replicates that the user wishes to specify
+
+        Returns                                                                                                                                          
+        -------                                                                                                                                          
+        est : ndarray                                                                                                                                    
+            The point estimates of the statistic, calculated on the columns                                                               
+            of data.                                                                                                                                     
+        vc : ndarray                                                                                                                              
+            The variance-covariance of the estimates.                                                                                                            
+        pseudo : ndarray                                                                                                                                 
+            The jackknife pseudo-values.
+        """
+        jdata = []
+        for r in range(replicates):
+            w = self.design.weights.copy()
+            bin = np.zeros(self.design.nclust)
+            for s in range(nstrat):
+                # how to handle strata w/ only one cluster?
+                w[self.design.strat == s] *= self.design.ncs[s] / float(self.design.ncs[s] - 1)
+                
+                # if there is only one or two clusters then weights will stay the same
+                if (self.design.ncs[s] == 1 or self.design.ncs[s] == 2):
+                    continue              
+                # array of clusters to resample from
+                ii = np.flatnonzero(self.design.sfclust == s)
+                # resample them
+                ii_resample = np.random.choice(ii, size = (self.design.ncs[s] - 1), replace=True)
+                # accumulate number of times cluster i was resampled
+                bin += np.bincount(ii_resample, minlength=max(self.design.sclust)+1)
+            # augment weights
+            w *= bin[self.design.sclust]
+            # call the stat w/ the new weights
+            jdata.append(stat._stat(weights=w))
+        jdata = np.asarray(jdata)
+        # nh = self.design.ncs[self.design.sfclust].astype(np.float64)
+        # pseudo = jdata + nh[:, None] * (np.dot(w, stat.data) - jdata)
+
+        boot_mean = jdata.mean(0)
+        var = ((jdata - boot_mean)**2).sum(0) / (replicates - 1)
+        est = stat._stat(self.design.weights)
+        return est, var
+
+    def jack(self, stat):
```

Comment:
It shouldn't be necessary to pass in stat as an argument.  The `_stat` method will be implemented in the subclasses, as you have done.